mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -208,6 +208,7 @@ int Conv2dGenerator::getKernelCount() const {
   case miopen::Conv2DBwdWeightOpType:
     return 1;
   }
+  llvm_unreachable("Invalid conv2d operation");
```

Comment:
A couple cases where compiler can't see that it can't legally fall through the switch, and warns that a value-returning function may not return a value.

Reply:
... yeah, Clang can do that reasoning, but GCC 7 can't

Reply:
At first I put the "unreachable" into a default case, but the compiler warned that I was using a default when all enumerations were represented.  Well, if it can determine that, why couldn't it determine that it would never proceed past the switch?  Foo.

Reply:
Might be a weird compiler bug thing.